src/app/routes/getInitialData/index.js

```diff
@@ -18,13 +20,19 @@ const baseUrl = onClient()
 
 const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
 
-const handleResponse = preprocessorRules => async response => {
+const getPageData = async json => {
+  const response = await json;
+  const { type } = pathOr({}, ['metadata'], response);
+  return preprocess(response, getPreprocessorRules(type));
+};
+
+const handleResponse = () => async response => {
```

Comment:
What about keeping the getPageData stuff in the response handling function like so:

```js
const handleResponse = () => async response => {
  const { status } = response;

  if (status === STATUS_CODE_OK) {
    const pageData = await response.json();
    const { type } = path(['metadata'], pageData);
    const processedPageData = preprocess(response, getPreprocessorRules(type));

    return {
      status,
      pageData: processedPageData,
    };
  }

  return { status };
};
```